pkg/cmd/pulumi/packagecmd/package_info.go

```diff
@@ -0,0 +1,283 @@
+		}
+		if split[1] != moduleName {
+			continue
+		}
+		resources[split[2]] = spec
+	}
+
+	if len(resources) == 0 {
+		return fmt.Errorf("module %q not found", moduleName)
+	}
+	fmt.Fprintf(stdout, "Resources: %d\n", len(resources))
+
+	fmt.Fprintln(stdout)
+	for _, name := range maputil.SortedKeys(resources) {
+		fmt.Fprintf(stdout, " - %s: %s\n", name, summaryFromDescription(resources[name].Description))
+	}
+	return nil
+}
+
+func showResourceInfo(spec *schema.PackageSpec, moduleName, resourceName string, stdout io.Writer) error {
+	fullResourceName := fmt.Sprintf("%s:%s:%s", spec.Name, moduleName, resourceName)
+
+	fmt.Fprintf(stdout, "Resource: %s\n", fullResourceName)
+
+	res, ok := spec.Resources[fullResourceName]
+	if !ok {
+		for name, r := range spec.Resources {
+			simplifiedName, err := simplifyModuleName(name)
+			if err != nil {
+				return err
+			}
+			if fullResourceName == simplifiedName {
+				res = r
+				ok = true
+				break
+			}
+		}
+	}
+	if !ok {
+		return fmt.Errorf("resource %q not found", fullResourceName)
+	}
+
+	fmt.Fprintln(stdout)
+	fmt.Fprintln(stdout, "Input Properties:")
+	for _, name := range maputil.SortedKeys(res.InputProperties) {
+		prop := res.InputProperties[name]
+		optionalStr := ""
+		if !slices.Contains(res.RequiredInputs, name) {
+			optionalStr = " (optional)"
+		}
+		typ, err := getType(spec, prop.TypeSpec)
+		if err != nil {
+			return err
+		}
+		fmt.Fprintf(stdout, " - %s (%s%s): %s\n",
+			name, typ, optionalStr, summaryFromDescription(prop.Description))
+	}
+
+	fmt.Fprintln(stdout)
+
+	fmt.Fprintln(stdout, "Output Properties:")
+	for _, name := range maputil.SortedKeys(res.Properties) {
+		prop := res.Properties[name]
+		presentStr := ""
+		if slices.Contains(res.Required, name) {
+			presentStr = " (always present)"
+		}
+		typ, err := getType(spec, prop.TypeSpec)
+		if err != nil {
+			return err
+		}
+		fmt.Fprintf(stdout, " - %s (%s%s): %s\n",
+			name, typ, presentStr, summaryFromDescription(prop.Description))
+	}
+	return nil
+}
+
+func getType(spec *schema.PackageSpec, prop schema.TypeSpec) (string, error) {
+	typ := prop.Type
+	if typ != "" && typ != "object" && typ != "array" {
+		return typ, nil
+	}
+	if prop.Type == "array" {
+		if prop.Items == nil {
+			return "[]unknown", nil
+		}
+		typ, err := getType(spec, *prop.Items)
+		if err != nil {
+			return "", err
+		}
+		return "[]" + typ, nil
+	}
+	if prop.Ref != "" {
+		if strings.HasPrefix(prop.Ref, "#/types/") {
+			ref := strings.TrimPrefix(prop.Ref, "#/types/")
+			ref = strings.ReplaceAll(ref, "%2F", "/")
+			if typeSpec, ok := spec.Types[ref]; ok {
+				if len(typeSpec.Enum) > 0 {
+					return fmt.Sprintf("enum(%s){%s}",
+						typeSpec.Type, formatEnumValues(typeSpec.Enum)), nil
+				}
+				simplifiedName, err := simplifyModuleName(ref)
+				if err != nil {
+					return "", err
+				}
+				split := strings.Split(simplifiedName, ":")
+				return split[2], nil
+			}
+		}
+		return prop.Ref, nil
+	}
+	if prop.Ref != "" {
```

Comment:
Duplicate of the above check?

Reply:
Yeah indeed, we never get here.  Removed.